### PR TITLE
feat(processors.parser): add merge with timestamp option

### PR DESCRIPTION
**Pull request by powersj**
By default, the merge processor will use the existing timestamp of the metric. This makes somesense when the user wants to keep the timestamp of when the metric was originally created. However, there are cases where if the user is also parsing data at this stage that they may want to set the timestamp as well.

Fixes: #13102

### CODE REVIEW

**Review by Hipska — APPROVED** [review]

File: plugins/processors/parser/parser.go [hunk 1]
```diff
@@ -120,6 +133,20 @@ func merge(base telegraf.Metric, metrics []telegraf.Metric) telegraf.Metric {
...
+func mergeWithTimestamp(base telegraf.Metric, metrics []telegraf.Metric) telegraf.Metric {
+	for _, metric := range metrics {
+		for _, field := range metric.FieldList() {
+			base.AddField(field.Key, field.Value)
+		}
+		for _, tag := range metric.TagList() {
+			base.AddTag(tag.Key, tag.Value)
+		}
+		base.SetName(metric.Name())
+		base.SetTime(metric.Time())
```
Comment (srebhan) [inline plugins/processors/parser/parser.go:145]: Shouldn't we catch the case where none of the given metrics contain a valid time and retain the original time? If not, we should at least document the behavior...

**Review by srebhan — COMMENTED** [review]
@powersj LGTM, just one question about the no-valid-time corner case...

File: plugins/processors/parser/parser.go (hunk 1 above)
Comment (powersj) [reply plugins/processors/parser/parser.go:145]: This is a great question. I started to work on a test case for this, each of the parsers I looked at call `time.Now()` and uses that if no timestamp is found. Meaning `metric.Time()` will always be non-zero.

Comment (srebhan) [reply plugins/processors/parser/parser.go:145]: Can we still wrap this with a
```golang
		if !m.Time().IsZero() {
			base.SetTime(metric.Time())
		}
```
just in case...

Comment (powersj) [reply plugins/processors/parser/parser.go:145]: sure :) done

**Review by srebhan — APPROVED** [review]
LGTM. Thanks @powersj!
